src/python-frontend/python_converter.cpp

```diff
@@ -345,26 +395,124 @@ void python_converter::get_conditional_stms(
   locationt location = get_location_from_decl(ast_node);
   then.location() = location;
 
-  // Create if code and append "then" block
+  // Create if or while code
   codet code;
   auto type = ast_node["_type"];
   if(type == "If")
     code.set_statement("ifthenelse");
   else if(type == "While")
     code.set_statement("while");
 
+  // Append "then" block
   code.copy_to_operands(cond, convert_expression_to_code(then));
 
-  // Append 'else' block to the statement
   if(ast_node.contains("orelse") && !ast_node["orelse"].empty())
   {
+    // Append 'else' block to the statement
     exprt else_expr = get_block(ast_node["orelse"]);
     code.copy_to_operands(convert_expression_to_code(else_expr));
   }
 
   target_block.copy_to_operands(code);
 }
 
+void python_converter::get_function_definition(
+  const nlohmann::json &function_node)
+{
+  // Function return type
+  code_typet type;
+  nlohmann::json return_node = function_node["returns"];
+  if(return_node.contains("id"))
+  {
+    type.return_type() = get_typet(return_node["id"].get<std::string>());
+  }
+  else if(return_node.contains("value") && return_node["value"].is_null())
+  {
+    type.return_type() = empty_typet();
+  }
+  else
+  {
+    log_error("Return type undefined\n");
+    abort();
+  }
+
+  // Function location
+  locationt location = get_location_from_decl(function_node);
+
+  // Symbol identification
+  current_func_name = function_node["name"].get<std::string>();
+  std::string id = "py:" + python_filename + "@F@" + current_func_name;
+  std::string module_name =
+    python_filename.substr(0, python_filename.find_last_of("."));
+
+  // Handle empty function arguments
+  if(function_node["args"]["args"].empty())
+  {
+    type.make_ellipsis();
```

Comment:
make_ellipsis() allows a function of that type to be called with an arbitrary number of more arguments than the type specifies. In C it's used to represent e.g.
```c
void f(int, ...)
```
Now f can be called as `f(42, 3, "more", "parameters")`. I don't think this is allowed for Python functions taking no arguments.

Reply:
Thanks for pointing that out.
Fixed.